lib/paper_trail/has_paper_trail.rb

```diff
@@ -78,6 +78,19 @@ module InstanceMethods
       def paper_trail
         ::PaperTrail::RecordTrail.new(self)
       end
+
+      def paper_trail_type_name
+        if @paper_trail_type_name.nil?
+          type_column = self.class.inheritance_column
+          @paper_trail_type_name =
+            (respond_to?(type_column) ? send(type_column) : nil) ||
+            self.class.name
+          if @paper_trail_type_name == self.class.base_class.name
+            @paper_trail_type_name = ""
+          end
+        end
+        @paper_trail_type_name
+      end
```

Comment:
- Can this method be moved into `RecordTrail`?
- Why use an instance variable? I'd prefer a local variable if possible.

Reply:
This caches @paper_trail_type_name -- although perhaps people would not often call .versions more than once on a versioned object.

If it moves anywhere then I'd prefer to have it back in model_config where the has_many is established.

(TBH this started because the Rubocop ABC is set at 22, and I couldn't get under 22.69 after breaking out the has_many stuff in model_config, and with `type_name` only being a simple string it seemed like a fairly lightweight addition to each object.)